### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by schoenkaft**
What's Up, Doc? 🐰 

To do:

- [x] Is the grouping of functions right? We shouldn't mix it all together I'd say...
- [x] Check `debounce` and `throttle`... should we return promises for both? Note to self: find easiest example possible, and re-check function call with promise and without since they might differ a bit...
- [x] Check if all examples are correct, might contain a stupid error or over-simplification here and there...

Feedback is welcome, but any commit/fixup will do as well, as this was only a Saturday-morning-lets-do-it-now-project 😄

### CODE REVIEW

**Comment by schoenkaft** [comment]
@harmenjanssen could you check the docs for clarity and errors? And if the grouping makes sense?

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+```js
+pick(['a', 'c'], { a: 1, b: 2, c: 3, d: 4 }); //=> { a: 1, c: 3 }
+```
+
+### preventDefault
+
+Call preventDefault on an event object. Might be combined with `tap` to create a chainable utility.
+
+See also `preventingDefault`.
```
Comment (harmenjanssen) [inline docs/README.md]:
```suggestion
See also [preventingDefault](#preventingDefault).
```

**Comment by harmenjanssen** [comment]
What so you think of this?
**Functional** is strictly function decorators, and I've added a section about object/array manipulation (which could be split further I guess).

```
## Objects and arrays

- [head](#head)
- [last](#last)
- [map](#map)
- [not](#not)
- [omit](#omit)
- [pick](#pick)
- [prop](#prop)
- [reverse](#reverse)
- [tail](#tail)
- [take](#take)
- [takeLast](#takeLast)

## Functional

- [compose](#compose)
- [curry](#curry)
- [debounce](#debounce)
- [memoize](#memoize)
- [tap](#tap)
- [throttle](#throttle)

## DOM / Browser

- [closest](#closest)
- [htmlToElement](#htmlToElement)
- [preventDefault](#preventDefault)
- [preventingDefault](#preventingDefault)
- [unescapeHtml](#unescapeHtml)

## Misc

- [defined](#defined)
- [parseJson](#parseJson)
- [uuid](#uuid)
```

**Comment by schoenkaft** [comment]
> What so you think of this?
> **Functional** is strictly function decorators, and I've added a section about object/array manipulation (which could be split further I guess).
> 
> [...]

Yeah! Was actually going to counter-propose something similar 🎉

**Pull request merged**